std/signal/test.ts

```diff
@@ -58,4 +58,38 @@ if (Deno.build.os !== "win") {
       await delay(10);
     },
   });
+
+  test({
+    name: "onSignal() registers and disposes of event handler",
+    async fn() {
+      // This prevents the program from exiting.
+      const t = setInterval(() => {}, 1000);
+
+      let calledCount = 0;
+      const handle = onSignal(Deno.Signal.SIGINT, () => {
+        calledCount++;
+      });
+
+      await delay(20);
+      Deno.kill(Deno.pid, Deno.Signal.SIGINT);
+      await delay(20);
+      Deno.kill(Deno.pid, Deno.Signal.SIGINT);
+      await delay(20);
+      Deno.kill(Deno.pid, Deno.Signal.SIGUSR2);
+      await delay(20);
+      handle.dispose(); // stop monitoring SIGINT
+      await delay(20);
+      Deno.kill(Deno.pid, Deno.Signal.SIGUSR1);
+      await delay(20);
+      Deno.kill(Deno.pid, Deno.Signal.SIGINT);
+      await delay(20);
+      assertEquals(calledCount, 2);
+
+      clearTimeout(t);
+      // Clear timeout clears interval, but interval promise is not
+      // yet resolved, delay to next turn of event loop otherwise,
+      // we'll be leaking resources.
+      await delay(10);
```

Comment:
@bartlomieju @piscisaureus we should fix this...

Reply:
Fixed in #4602

Reply:
Ah  great - then these delays can be removed?

Reply:
Yes, it should be possible - I already did that for other signal tests

Reply:
Addressed in #4703